src/test/java/SalesforceModules/QuoteUtil.java

```diff
@@ -1037,28 +1053,208 @@ public static void changeStatusAsDraft_ss(String status, String logStep) throws
+     *
+     * @param logStep the log
+     */
+    public static void newButtonQuote_billing(String logStep) throws InterruptedException {
+        KeywordUtil.delay(4000);
+        WebElement element = KeywordUtil.getDriver().findElement(QuoteObject.clickOnNew);
+        JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
+        executor.executeScript("arguments[0].click();", element);
+    }
+
+    /**
+     * Select the Account field for new quote in billing section
+     *
+     * @param logStep the log
+     * @param account the account name
+     */
+    public static void selectAccount_billing(String account, String logStep) throws InterruptedException {
+        boolean flag = false;
+        KeywordUtil.waitForVisible(ServiceSupportObject.searchAccounts_quote_ss);
+        KeywordUtil.click(ServiceSupportObject.searchAccounts_quote_ss, logStep);
+        KeywordUtil.inputText(ServiceSupportObject.searchAccounts_quote_ss, "TX", logStep);
+        KeywordUtil.delay(5000);
+        Thread.sleep(5000);
+        String accname = "(//lightning-base-combobox-formatted-text[contains(@title,'TX')])[1]";
+
+        try {
+            flag = KeywordUtil.getDriver().findElement(By.xpath(accname)).isDisplayed();
+
+        } catch (Exception e) {
+        }
+
+        if (!flag) {
+            System.out.println("no such account present");
+        } else {
+            KeywordUtil.click(By.xpath(accname), "account selected");
+        }
+    }
+
+    /**
+     * Click EditLines for billing
+     *
+     * @param logStep the log
+     */
+    public static void clickEditLines_billing(String logStep) throws InterruptedException {
+        KeywordUtil.delay(10000);
+        WebElement element = KeywordUtil.getDriver().findElement(QuoteObject.dropdownInQuotesTab);
+        JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
+        executor.executeScript("arguments[0].click();", element);
+        KeywordUtil.delay(10000);
+        KeywordUtil.waitForClickable(QuoteObject.editLines_b);
+        KeywordUtil.click(QuoteObject.editLines_b, logStep);
+    }
+
+    /**
+     * Click edit lines and add discount for billing section
+     */
+    public static void addDiscountProduct_billing() throws InterruptedException {
+        clickEditLines_billing("Edit Lines clicked");
+        enterDiscount_billing("5", "2FAUSBKEY", "Entered discount");
+    }
+
+    /**
+     * Switch to IFrame
+     * Add discount for billing section
+     */
+
+    public static void enterDiscount_billing(String discount, String product, String logStep) {
+        try {
+
+            JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
+            WebElement iframe = GlobalUtil.getDriver().findElement(By.xpath("//div[@class=\"windowViewMode-normal oneContent active lafPageHost\"]//iframe"));
+            GlobalUtil.getDriver().switchTo().frame(iframe);
```

Comment:
why are not we using the reusable method to switch to frame?

Reply:
ok